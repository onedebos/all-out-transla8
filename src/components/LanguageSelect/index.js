import styled from 'styled-components';

export default function LanguageSelect({ langType, langOptions, onChange }) {
	return (
		<SelectWrapper>
			<LanguageSelectStyle aria-label="select-language" name={langType} data-testid={langType} onChange={onChange}>
				{langOptions.map((srcLang, index) => (
					<LangStyleOption data-testid={langType + '-option'} key={index} value={srcLang.langCode}>
						{srcLang.lang}
					</LangStyleOption>
				))}
			</LanguageSelectStyle>
		</SelectWrapper>
	);
}

const SelectWrapper = styled.div`
	position: relative;
	max-width: 300px;
	margin-right: 20px;

	&::after {
		content: 'â–¼';
		font-size: 1rem;
		top: 9px;
		right: 15px;
		position: absolute;
		color: #1a73e8;
	}
`;

const LanguageSelectStyle = styled.select`
	border: 1px solid #dfe1e5;
	color: #1a73e8;
	border-radius: 8px;
	padding-left: 18px;
	height: 40px;
	font-size: 14px;
	width: 300px;
	-webkit-appearance: none;
	appearance: none;
`;

const LangStyleOption = styled.option``;
