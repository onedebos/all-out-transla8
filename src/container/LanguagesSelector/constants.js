export default {
	srcLangOptions: [{ langCode: 'EN', lang: 'English' }],

	targetLangOptions: [
		{ langCode: 'BG', lang: 'Bulgarian' },
		{ langCode: 'CS', lang: 'Czech' },
		{ langCode: 'DA', lang: 'Danish' },
		{ langCode: 'DE', lang: 'German' },
		{ langCode: 'EL', lang: 'Greek' },
		{ langCode: 'EN-GB', lang: 'English (British)' },
		{ langCode: 'EN-US', lang: 'English (American)' },
		// "EN" - English (unspecified variant for backward compatibility; please select EN-GB or EN-US instead)
		{ langCode: 'ES', lang: 'Spanish' },
		{ langCode: 'ET', lang: 'Estonian' },
		{ langCode: 'FI', lang: 'Finnish' },
		{ langCode: 'FR', lang: 'French' },
		{ langCode: 'HU', lang: 'Hungarian' },
		{ langCode: 'IT', lang: 'Italian' },
		{ langCode: 'JA', lang: 'Japanese' },
		{ langCode: 'LT', lang: 'Lithuanian' },
		{ langCode: 'LV', lang: 'Latvian' },
		{ langCode: 'NL', lang: 'Dutch' },
		{ langCode: 'PL', lang: 'Polish' },
		{ langCode: 'PT-PT', lang: 'Portuguese' },
		{ langCode: 'PT-BR', lang: 'Portuguese (Brazilian)' },
		// {"PT" : Portuguese (unspecified variant for backward compatibility; please select PT:PT or PT:BR instead)}
		{ langCode: 'RO', lang: 'Romanian' },
		{ langCode: 'RU', lang: 'Russian' },
		{ langCode: 'SK', lang: 'Slovak' },
		{ langCode: 'SL', lang: 'Slovenian' },
		{ langCode: 'SV', lang: 'Swedish' },
		{ langCode: 'ZH', lang: 'Chinese' },
	],

	SRC_LANG: 'Source Language',
	TRGT_LANG: 'Target Language',
};
