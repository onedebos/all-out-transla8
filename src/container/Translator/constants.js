export default {
    NO_SRC_TEXT_ERR: "Enter a word to translate...",
    NO_TRGT_LANG_ERR: "Select a language to translate into."
}