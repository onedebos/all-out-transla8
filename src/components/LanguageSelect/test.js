/**
 * @jest-environment jsdom
 */
 import { render, screen, cleanup } from '@testing-library/react';
 import '@testing-library/jest-dom';
 import renderer from 'react-test-renderer';
 import LanguageSelect from './';
 import constants from '../../container/LanguagesSelector/constants'

 const {srcLangOptions, SRC_LANG,  TRGT_LANG, targetLangOptions} = constants
 
 afterEach(() => {
     cleanup();
 });
 
 describe('<Button />', () => {
     it('renders the LanguageSelect component successfully without crashing', () => {
         render(<LanguageSelect langType={SRC_LANG} langOptions={srcLangOptions} />);
     });

     it('renders only shows English as an option in the source language dropdown', () => {
        render(<LanguageSelect langType={SRC_LANG} langOptions={srcLangOptions} />);
        const langOptions = screen.getByTestId('lang-options');
		expect(langOptions).toBeInTheDocument();
        expect(langOptions).toHaveTextContent('English');
        expect(langOptions).not.toHaveTextContent('Bulgarian');
    });

    it('renders several languages as options in the target language dropdown', () => {
        render(<LanguageSelect langType={TRGT_LANG} langOptions={targetLangOptions} />);
        const langOptions = screen.getByTestId('lang-options');
		expect(langOptions).toBeInTheDocument();
        expect(langOptions).toHaveTextContent('English');
        expect(langOptions).toHaveTextContent('Bulgarian');
        expect(langOptions).toHaveTextContent('German');
        expect(langOptions).toHaveTextContent('French');
    });
 
     
     it('matches snapshot', () => {
         const tree = renderer.create(<LanguageSelect langType={TRGT_LANG} langOptions={targetLangOptions} />).toJSON();
         expect(tree).toMatchSnapshot();
     });
 });
 