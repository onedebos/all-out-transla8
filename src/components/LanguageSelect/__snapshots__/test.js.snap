// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageSelect /> matches snapshot 1`] = `
<div
  className="LanguageSelect__SelectWrapper-sc-1umrqq7-0 iZfAxW"
>
  <select
    aria-label="select-language"
    className="LanguageSelect__LanguageSelectStyle-sc-1umrqq7-1 nRspY"
    data-testid="TargetLanguage"
    name="TargetLanguage"
  >
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="BG"
    >
      Bulgarian
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="CS"
    >
      Czech
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="DA"
    >
      Danish
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="DE"
    >
      German
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="EL"
    >
      Greek
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="EN-GB"
    >
      English (British)
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="EN-US"
    >
      English (American)
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="ES"
    >
      Spanish
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="ET"
    >
      Estonian
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="FI"
    >
      Finnish
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="FR"
    >
      French
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="HU"
    >
      Hungarian
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="IT"
    >
      Italian
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="JA"
    >
      Japanese
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="LT"
    >
      Lithuanian
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="LV"
    >
      Latvian
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="NL"
    >
      Dutch
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="PL"
    >
      Polish
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="PT-PT"
    >
      Portuguese
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="PT-BR"
    >
      Portuguese (Brazilian)
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="RO"
    >
      Romanian
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="RU"
    >
      Russian
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="SK"
    >
      Slovak
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="SL"
    >
      Slovenian
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="SV"
    >
      Swedish
    </option>
    <option
      className="LanguageSelect__LangStyleOption-sc-1umrqq7-2"
      data-testid="TargetLanguage-option"
      value="ZH"
    >
      Chinese
    </option>
  </select>
</div>
`;
